File: src/python/handlers/tdays.py
from utils import wind_data_to_dict, handle_error
from WindPy import w


def handle_tdays(params: dict) -> dict:
    begin_time = params["beginTime"]
    end_time = params["endTime"]
    options = params.get("options", "")

    data = w.tdays(begin_time, end_time, options)

    if data.ErrorCode != 0:
        raise RuntimeError(handle_error(data.ErrorCode))

    return wind_data_to_dict(data)

File: src/tools/tdays.ts
import { z } from "zod";
import { runBridge } from "../bridge/runner.js";

export const tdaysTool = {
  name: "wind_tdays",
  description: "Get trading calendar days between two dates using Wind w.tdays().",
  inputSchema: z.object({
    beginTime: z.string().describe("Start date in YYYY-MM-DD format"),
    endTime: z.string().describe("End date in YYYY-MM-DD format"),
    options: z.string().optional().describe("Wind options string, e.g. 'Days=Alldays'"),
  }),
  handler: async (params: Record<string, unknown>) => {
    const result = await runBridge({ function: "tdays", params });
    if (!result.ok) throw new Error(result.error);
    return result.data;
  },
};
